ui/src/combobox.tsx

```diff
@@ -110,22 +111,23 @@ const ComboboxMultiSelect = ({ model: m }: { model: Omit<Combobox, 'value'> }) =
     [options, setOptions] = useOptions(m.choices),
     [selected, setSelected] = React.useState<S[]>(m.values ?? []),
     selectOpt = (option: IComboBoxOption) => {
-      setSelected(keys => {
-        const result = option.selected ? [...keys, String(option.key)] : keys.filter(key => key !== option.key)
-        wave.args[m.name] = result
-        return result
+      flushSync(() => {
+        setSelected(keys => {
+          const result = option.selected ? [...keys, String(option.key)] : keys.filter(key => key !== option.key)
+          wave.args[m.name] = result
+          return result
+        })
       })
+
+      if (m.trigger) wave.push()
```

Comment:
Can using the `flushSync` be avoided? This should be enough:
```suggestion
      const result = option.selected ? [...selected, String(option.key)] : selected.filter(key => key !== option.key)
      setSelected(result)
      wave.args[m.name] = result
      if (m.trigger) wave.push()
```

Reply:
Our `selected` state depends on its prev state thus a callback is necessary to make it safe.

flushSync is not risky, it just avoids batching state updates, but since we need to know the prev state, we do want sync updates anyway.

Reply:
Using your suggestion should work as well, but there are no guarantees from React that each user event like blur, tab, enter etc. commit state so I prefer to not depend on the internal implementation, especially when we are in the maintenance mode and I want each of us to spend at little time on this project as possible.

Reply:
Thanks for clarification! Your points make sense.